gym/vector/vector_env.py

```diff
@@ -211,6 +217,42 @@ def seed(self, seed=None):
             "Please use `env.reset(seed=seed) instead in VectorEnvs."
         )
 
+    def _add_info(self, infos: dict, info: dict, env_num: int) -> dict:
+        """Add env info to the info dictionary of the vectorized environment.
+
+        Given the `info` of a single environment add it to the `infos` dictionary
+        which represents all the infos of the vectorized environment.
+        Every `key` of `info` is paired with a boolean mask `_key` representing
+        whether or not the i-indexed environment has this `info`.
+
+        Args:
+            infos (dict): the infos of the vectorized environment
+            info (dict): the info coming from the single environment
+            env_num (int): the index of the single environment
+
+        Returns:
+            infos (dict): the (updated) infos of the vectorized environment
+
+        """
+        for k in info.keys():
+            if k not in infos:
+                info_array, array_mask = self._init_info_arrays(type(info[k]))
+            else:
+                info_array, array_mask = infos[k], infos[f"_{k}"]
+
+            info_array[env_num], array_mask[env_num] = info[k], True
+            infos[k], infos[f"_{k}"] = info_array, array_mask
+        return infos
+
+    def _init_info_arrays(self, dtype: type) -> np.ndarray:
+        if dtype in [int, float, bool] or issubclass(dtype, np.number):
+            array = np.zeros(self.num_envs, dtype=dtype)
```

Comment:
Can we add some explicit handling for numpy arrays too? I think the current implementation can cause some problems through a mixup between e.g. `np.float32` and `np.float64`, or it will even default to the `object` type. And I can easily see passing an array in the info dict as a desired functionality

Reply:
No problem for me, atm if an array is added to the info dict the resulting output will fall in the `object` dtype resulting in this:
```python
"terminal_observation": [
        array([0.2097086, -0.5355723, -0.21343598, -0.11173592], dtype=float32),
        None,
        None
    ]
```
Something I can think of is making the output in this form:
```python
"terminal_observation": [
        array(
            [[0.2097086, -0.5355723, -0.21343598, -0.11173592],
            [0., 0., 0., 0.],
            [0., 0., 0., 0.]],
           dtype=float32),
    ]
```
I'm not sure this adhere to the format of google-brax tho